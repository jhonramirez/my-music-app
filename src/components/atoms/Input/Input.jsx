export const Input = ({ type, id, defaultChecked, onClick }) => {
  return (
    <input
      type={type}
      id={id}
      defaultChecked={defaultChecked}
      onClick={onClick}
    />
  );
};
