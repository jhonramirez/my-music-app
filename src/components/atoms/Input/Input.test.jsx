import { fireEvent, render, screen } from "@testing-library/react";
import { Input } from "./Input";
import "@testing-library/jest-dom";

describe("Pruebas en <Input />", () => {
  const { container } = render(<Input id={"proof"} />);
  const InputElement = screen.getByTestId("heart-input");

  test("Render Input y match con el snapshot", () => {
    expect(InputElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("Se evalua que el Input tenga un ID", () => {
    expect(InputElement).toHaveAttribute("id", "proof");
  });

  test("Se evalua que se ejecute el onClick", () => {
    const onClick = jest.fn();
    render(<Input id={"proof"} onClick={onClick} />);
    const InputElement = screen.getByTestId("heart-input");
    fireEvent.click(InputElement);
    expect(onClick).toHaveBeenCalled();
  });
});
