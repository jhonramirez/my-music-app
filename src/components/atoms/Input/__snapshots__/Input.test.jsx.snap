// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Input /> Render Input y match con el snapshot 1`] = `
<div>
  <input
    data-testid="heart-input"
    id="proof"
  />
</div>
`;
